### ISSUE TITLE
A lot of crashes in  -[BugsnagBreadcrumb setType:] (BugsnagBreadcrumb.m:135:5)

### ISSUE THREAD

**Issue by Naugladur** [bugsnag/bugsnag-cocoa#689]
### Description
I get a large number of errors in my project when the `-[BugsnagBreadcrumb setType:]` method works.

The implementation of breadcrumbs is completely default, application don't call any methods to use breadcrumbs.

### Environment

Library versions:

Bugsnag version (from your Podfile, Podfile.lock, or elsewhere): 5.23.2
CocoaPods version (if any) (pod -v): -
Carthage version (if any): -
iOS/tvOS/macOS version(s): iOS 13+
debug mode or production?: both
How are you initializing Bugsnag? Can you share a snippet of your
AppDelegate where Bugsnag is configured?
Integrated via subproject in workspace.

```swift
func setup() {
    let config = BugsnagConfiguration()
    config.apiKey = "Key"
    config.releaseStage = "production"
    Bugsnag.start(with: config)
}
```

### Error messages

```
EXC_BREAKPOINT: 

0  libsystem_platform.dylib __os_unfair_lock_corruption_abort
1  libsystem_platform.dylib __os_unfair_lock_lock_slow
2  libobjc.A.dylib          _objc_sync_enter
3  app                      -[BugsnagBreadcrumb setType:] (BugsnagBreadcrumb.m:135:5)
4  app                      __38-[BugsnagNotifier orientationChanged:]_block_invoke (BugsnagNotifier.m:796:22)
5  app                      +[BugsnagBreadcrumb breadcrumbWithBlock:] (BugsnagBreadcrumb.m:169:9)
6  app                      -[BugsnagBreadcrumbs addBreadcrumbWithBlock:] (BugsnagBreadcrumb.m:211:32)
7  app                      -[BugsnagNotifier addBreadcrumbWithBlock:] (BugsnagNotifier.m:695:5)
8  app                      -[BugsnagNotifier orientationChanged:] (BugsnagNotifier.m:795:9)
9  Foundation               ___57-[NSNotificationCenter addObserver:selector:name:object:]_block_invoke_2
10 CoreFoundation           ___CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__
11 CoreFoundation           ____CFXRegistrationPost1_block_invoke
12 CoreFoundation           __CFXRegistrationPost1
13 CoreFoundation           ____CFXNotificationPost_block_invoke
14 CoreFoundation           -[_CFXNotificationRegistrar find:object:observer:enumerator:]
15 CoreFoundation           __CFXNotificationPost
16 Foundation               -[NSNotificationCenter postNotificationName:object:userInfo:]
17 UIKitCore                -[UIDevice setOrientation:animated:]

{
  "code": 3101697512,
  "exception_name": "EXC_BREAKPOINT",
  "subcode": 8,
  "exception": 6
}
```

<details><summary>Full error message:</summary>

```
0  libsystem_platform.dylib __os_unfair_lock_corruption_abort
1  libsystem_platform.dylib __os_unfair_lock_lock_slow
2  libobjc.A.dylib          _objc_sync_enter
3  app                      -[BugsnagBreadcrumb setType:] (BugsnagBreadcrumb.m:135:5)
4  app                      __38-[BugsnagNotifier orientationChanged:]_block_invoke (BugsnagNotifier.m:796:22)
5  app                      +[BugsnagBreadcrumb breadcrumbWithBlock:] (BugsnagBreadcrumb.m:169:9)
6  app                      -[BugsnagBreadcrumbs addBreadcrumbWithBlock:] (BugsnagBreadcrumb.m:211:32)
7  app                      -[BugsnagNotifier addBreadcrumbWithBlock:] (BugsnagNotifier.m:695:5)
8  app                      -[BugsnagNotifier orientationChanged:] (BugsnagNotifier.m:795:9)
9  CoreFoundation           ___CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__
10 CoreFoundation           ____CFXRegistrationPost1_block_invoke
11 CoreFoundation           __CFXRegistrationPost1
12 CoreFoundation           ____CFXNotificationPost_block_invoke
13 CoreFoundation           -[_CFXNotificationRegistrar find:object:observer:enumerator:]
14 CoreFoundation           __CFXNotificationPost
15 Foundation               -[NSNotificationCenter postNotificationName:object:userInfo:]
16 UIKitCore                -[UIDevice setOrientation:animated:]
17 UIKitCore                ___134-[_UIWindowSceneDeviceOrientationSettingsDiffAction _updateDeviceOrientationWithSettingObserverContext:windowScene:transitionContext:]_block_invoke
18 UIKitCore                +[BSAnimationSettings(UIKit) tryAnimatingWithSettings:actions:completion:]
19 UIKitCore                __UISceneSettingsDiffActionPerformChangesWithTransitionContext
20 UIKitCore                -[_UIWindowSceneDeviceOrientationSettingsDiffAction _updateDeviceOrientationWithSettingObserverContext:windowScene:transitionContext:]
21 UIKitCore                ___163-[_UIWindowSceneDeviceOrientationSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke
22 UIKitCore                __UISceneSettingsDiffActionPerformActionsWithDelayForTransitionContext
23 UIKitCore                -[_UIWindowSceneDeviceOrientationSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]
24 UIKitCore                ___64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke
25 UIKitCore                -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:]
26 UIKitCore                -[UIScene scene:didUpdateWithDiff:transitionContext:completion:]
27 UIKitCore                -[UIApplicationSceneClientAgent scene:handleEvent:withCompletion:]
28 FrontBoardServices       -[FBSSceneImpl updater:didUpdateSettings:withDiff:transitionContext:completion:]
29 FrontBoardServices       ___88-[FBSWorkspaceScenesClient sceneID:updateWithSettingsDiff:transitionContext:completion:]_block_invoke_2
30 FrontBoardServices       -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:]
31 FrontBoardServices       ___88-[FBSWorkspaceScenesClient sceneID:updateWithSettingsDiff:transitionContext:completion:]_block_invoke
32 libdispatch.dylib        __dispatch_client_callout
33 libdispatch.dylib        __dispatch_block_invoke_direct$VARIANT$armv81
34 FrontBoardServices       __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__
35 FrontBoardServices       -[FBSSerialQueue _queue_performNextIfPossible]
36 FrontBoardServices       -[FBSSerialQueue _performNextFromRunLoopSource]
37 CoreFoundation           ___CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
38 CoreFoundation           ___CFRunLoopDoSource0
39 CoreFoundation           ___CFRunLoopDoSources0
40 CoreFoundation           ___CFRunLoopRun
41 CoreFoundation           _CFRunLoopRunSpecific
42 GraphicsServices         _GSEventRunModal
43 UIKitCore                _UIApplicationMain
44 app                      main (main.swift:12:1)
45 libdyld.dylib            _start
```

</details>

**Comment by johnkiely1** [comment]
Hi @Naugladur 

It would be worth upgrading to [v5.23.3](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v5.23.3) as there was a locking issue identified in v5.23.2 which could be related. This was fixed in v5.23.3

Once this is complete, if the error still persists, could you try adding `config.automaticallyCollectBreadcrumbs = false` which should effectively turn off all automatic breadcrumbs. It would be helpful to know if this stops the lock from occurring. 
 
Thanks

**Comment by johnkiely1** [comment]
Hi @Naugladur,
On second thought it looks like the lock issue fixed in 5.23.3 is not the same as what you are experiencing so my previous advice to upgrade can be ignored. 

Turning off the breadcrumbs is still worth trying to help pinpoint the cause.

It would however be useful for us if you could email in to support@bugsnag.com with details of the project and give us the link to the error as it appears in your Bugsnag dashboard.

Thanks

**Comment by johnkiely1** [comment]
As this has been continued as a support request I am now closing this ticket, we will update this thread with any relevant information when available.
